### PR TITLE
Fix note about cache limitations

### LINKED ISSUE #2271: Caching on self-hosted runners
### What article on docs.github.com is affected?
https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-circleci-to-github-actions#caching

### What part(s) of the article would you like to see updated?
Caching

### Additional information
The docs state that `GitHub Actions caching is only applicable to GitHub-hosted runners.` which seems to indicate we can't use caching on self-hosted runners, but I just tested caching on a self-hosted runner and it worked fine. Are the docs out of date?

### PR DESCRIPTION
**Pull request by Maks3w**
### Why:

Fix #2271

### What's being changed:

As discussed in https://github.com/github/docs/issues/2271#issuecomment-785479123

### Check off the following:
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: content/actions/learn-github-actions/migrating-from-circleci-to-github-actions.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ GitHub Actions
 </tr>
 </table>
 
-{% data variables.product.prodname_actions %} caching is only applicable to {% data variables.product.prodname_dotcom %}-hosted runners. For more information, see "<a href="/actions/guides/caching-dependencies-to-speed-up-workflows" class="dotcom-only">Caching dependencies to speed up workflows</a>."
+{% data variables.product.prodname_actions %} caching is only applicable for repositories hosted on {% data variables.product.prodname_dotcom %}.com. For more information, see "<a href="/actions/guides/caching-dependencies-to-speed-up-workflows" class="dotcom-only">Caching dependencies to speed up workflows</a>."
```
Comment (janiceilene) [inline content/actions/learn-github-actions/migrating-from-circleci-to-github-actions.md:110]:
```suggestion
{% data variables.product.prodname_actions %} caching is only applicable for repositories hosted on {% data variables.product.prodname_dotcom_the_website %}. For more information, see "<a href="/actions/guides/caching-dependencies-to-speed-up-workflows" class="dotcom-only">Caching dependencies to speed up workflows</a>."
```
Thanks so much for following up with a PR @Maks3w! I think using this variable from `data/variables/product.yml` should render correctly. Let me know when you commit this suggestion and I'll get this merged ✨

Comment (Maks3w) [reply content/actions/learn-github-actions/migrating-from-circleci-to-github-actions.md]: Done. I suspect some variable should exists but I think it was on the index were it was composed with the `.com` sufix

**Comment by janiceilene** [comment]
Thanks @Maks3w! I'll udpate this branch and get it merged once tests are passing 💖

**Review by janiceilene — APPROVED** [review]

File: content/actions/learn-github-actions/migrating-from-circleci-to-github-actions.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ GitHub Actions
 </tr>
 </table>
 
-{% data variables.product.prodname_actions %} caching is only applicable to {% data variables.product.prodname_dotcom %}-hosted runners. For more information, see "<a href="/actions/guides/caching-dependencies-to-speed-up-workflows" class="dotcom-only">Caching dependencies to speed up workflows</a>."
+{% data variables.product.prodname_actions %} caching is only applicable for repositories hosted on {% data variables.product.prodname_dotcom_the_website %}. For more information, see "<a href="/actions/guides/caching-dependencies-to-speed-up-workflows" class="dotcom-only">Caching dependencies to speed up workflows</a>."
```
Comment (ssabawissabawi) [inline content/actions/learn-github-actions/migrating-from-circleci-to-github-actions.md:120]: [image](https://github.com/github/docs/commit/2be566ca3e83cba9f3149f0a4fe28386a5a09885)

Comment (ssabawissabawi) [inline content/actions/learn-github-actions/migrating-from-circleci-to-github-actions.md:120]: _i_

**Review by ssabawissabawi — COMMENTED** [review]
https://github.com/github/docs/commit/2be566ca3e83cba9f3149f0a4fe28386a5a09885j

**Pull request merged**